core/src/main/java/org/lflang/target/property/TracingProperty.java

```diff
@@ -0,0 +1,169 @@
+package org.lflang.target.property;
+
+import java.util.List;
+import java.util.Objects;
+import org.lflang.AbstractTargetProperty;
+import org.lflang.MessageReporter;
+import org.lflang.Target;
+import org.lflang.ast.ASTUtils;
+import org.lflang.lf.Element;
+import org.lflang.lf.KeyValuePair;
+import org.lflang.lf.KeyValuePairs;
+import org.lflang.lf.LfFactory;
+import org.lflang.lf.LfPackage.Literals;
+import org.lflang.lf.Model;
+import org.lflang.target.TargetProperty;
+import org.lflang.target.property.TracingProperty.TracingOptions;
+import org.lflang.target.property.type.DictionaryType;
+import org.lflang.target.property.type.DictionaryType.DictionaryElement;
+import org.lflang.target.property.type.PrimitiveType;
+import org.lflang.target.property.type.TargetPropertyType;
+import org.lflang.target.property.type.UnionType;
+
+public class TracingProperty extends AbstractTargetProperty<TracingOptions> {
+
+  public TracingProperty() {
+    super(UnionType.TRACING_UNION);
+  }
+
+  @Override
+  public TracingOptions initialValue() {
+    return new TracingOptions(false);
+  }
+
+  @Override
+  public TracingOptions fromAst(Element node, MessageReporter reporter) {
```

Comment:
There is a challenge in properties like tracing. In C++, the tracing option is simply considered to be a boolean. In C, however, there is additional configuration possible (i.e. specifying a file). As far as I understand the code now, there would be no warning or error when specifying a trace file in C++ (which was probably already the case before). But how can we handle these target specific sub properties?

We have the same problem with the platform property. I think it would make sense to support this one also in C++ (and remove the dedicated `ros2` property), but all the platform options like baudrate are quite useless in this context.

It seems like the "Options" of a property, should also be properties. We could parse and validate them nested.
It seems like the "Options" should also be properties

Reply:
That's an interesting thought... We do recursively check types, but not whether the current target has support... I think that if we do this, we should rely on reflection to figure out what "options" exist. If we go down this path, this would also remove the need for the `TargetProperty` enum...

Reply:
I made progress toward this in #2035, as it does proper recursive checking, but a full generalization like the one you describe is a bridge too far, I think. Because all of the structural checking is done automatically through the type classes, all that needs to happen in the `validate` method of the target property is to do additional checks (such as cross checks between target properties). Hence, if we want an error message when the dictionary type is used in the C++ target, we can easily handle this corner case in the `TracingProperty.validate`.

Reply:
But then we need to make `TracingPolicy.validate` aware of the target and have target specific validation code. This is reversed in the same way as the `isSupported` property as [pointed out by Clément](https://github.com/lf-lang/lingua-franca/pull/2008#discussion_r1345934810).

Reply:
This is a good point, but I don't know how to fix this without spending another two weeks on this :sweat_smile: I'm happy to explore fixing this at a later time. All of the changes in these PRs are steps in the right direction already...